tests/operators/test_dataframe.py

```diff
@@ -1,287 +1,223 @@
-        def my_df_func(df: pandas.DataFrame):
-            return df.actor_id.count()
-
-        res = self.create_and_run_task(
-            my_df_func,
-            (
-                Table(
-                    "actor", conn_id="postgres_conn", database="pagila", schema="public"
-                ),
-            ),
-            {},
-        )
-        assert (
-            XCom.get_one(
-                execution_date=DEFAULT_DATE, key=res.key, task_id=res.operator.task_id
-            )
-            == 200
-        )
-
-    def test_dataframe_from_sql_basic_op_arg_and_kwarg(self):
-        @df(conn_id="postgres_conn", database="pagila")
-        def my_df_func(actor_df: pandas.DataFrame, film_df: pandas.DataFrame):
-            return actor_df.actor_id.count() + film_df.film_id.count()
-
-        res = self.create_and_run_task(
-            my_df_func,
-            (
-                Table(
-                    "actor", conn_id="postgres_conn", database="pagila", schema="public"
-                ),
-            ),
-            {
-                "film_df": Table(
-                    "film", conn_id="postgres_conn", database="pagila", schema="public"
-                )
+        }
+    ],
+    indirect=True,
+)
+def test_dataframe_from_sql_basic(sample_dag, sql_server, test_table):
+    sql_name, hook = sql_server
+
+    @df
+    def my_df_func(df: pandas.DataFrame):
+        return df.sell.count()
+
+    with sample_dag:
+        f = my_df_func(df=test_table)
+
+    test_utils.run_dag(sample_dag)
+
+    assert (
+        XCom.get_one(execution_date=DEFAULT_DATE, key=f.key, task_id=f.operator.task_id)
+        == 5
+    )
+
+
+@pytest.mark.parametrize("sql_server", SUPPORTED_DATABASES, indirect=True)
+@pytest.mark.parametrize(
+    "test_table",
+    [
+        {
+            "path": str(CWD) + "/../data/homes2.csv",
+            "load_table": True,
+            "is_temp": False,
+            "param": {
+                "schema": SCHEMA,
+                "table_name": test_utils.get_table_name("test_stats_check_2"),
             },
-        )
-        assert (
-            XCom.get_one(
-                execution_date=DEFAULT_DATE, key=res.key, task_id=res.operator.task_id
-            )
-            == 1200
-        )
-
-    def test_snow_dataframe_from_sql_basic(self):
-        @df(identifiers_as_lower=False)
-        def my_df_func(df: pandas.DataFrame):
-            return df.LIVING.count()
-
-        res = self.create_and_run_task(
-            my_df_func,
-            (),
-            {
-                "df": Table(
-                    self.OUTPUT_TABLE_NAME,
-                    conn_id="snowflake_conn",
-                    schema=os.getenv("SNOWFLAKE_SCHEMA"),
-                    database=os.getenv("SNOWFLAKE_DATABASE"),
-                    warehouse=os.getenv("SNOWFLAKE_WAREHOUSE"),
-                )
+        }
+    ],
+    indirect=True,
+)
+def test_dataframe_from_sql_custom_task_id(sample_dag, sql_server, test_table):
```

Comment:
Are we testing that tasks with `custom_task_id` runs fine? This feels like a case where we don't need to run the dag for this case, only assess if the `task_id` is applied correctly or not>

Testing dag can be run or not seems redundant, please correct me if I am wrong

Reply:
@utkarsharma2 ^^

Reply:
I think this is test is to ensure we don't generate duplicate task_id. I see your point, the `assert` itself is before the run_dag() function. Also, I think we can get rid of the last `assert` statement since that functionality is tested in other test cases.